spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java

```diff
@@ -1304,6 +1313,82 @@ public void testDeleteToCustomWapBranchWithoutWhereClause() throws NoSuchTableEx
         });
   }
 
+  @Test
+  public void testDeleteFilterWithSystemFuncOnPartitionedTable() {
+    Assumptions.assumeThat(catalogName).isNotEqualTo("spark_catalog");
+
+    Map<String, String> tableProperties = Maps.newHashMap();
+    tableProperties.put(DEFAULT_FILE_FORMAT, fileFormat);
+    tableProperties.put(WRITE_DISTRIBUTION_MODE, distributionMode);
+    tableProperties.put(SPARK_WRITE_PARTITIONED_FANOUT_ENABLED, String.valueOf(fanoutEnabled));
+    tableProperties.putAll(extraTableProperties());
+    createPartitionedTable(spark, tableName, "years(ts)", tableProperties);
+
+    String date = "2018-12-21";
+    String remainSql =
+        sqlFormat(
+            "SELECT * FROM %s WHERE %s.system.years(ts) != %s.system.years(date('%s')) ORDER BY id",
+            tableName, catalogName, catalogName, date);
+
+    List<Object[]> expected = sql(remainSql);
+
+    withSQLConf(
+        ImmutableMap.of(SparkSQLProperties.SYSTEM_FUNC_PUSH_DOWN_ENABLED, "true"),
+        () -> {
+          String deleteSql =
+              sqlFormat(
+                  "DELETE FROM %s WHERE %s.system.years(ts) == %s.system.years(date('%s'))",
+                  tableName, catalogName, catalogName, date);
+          Object explain = scalarSql("EXPLAIN EXTENDED %s", deleteSql);
+          // Metadata delete
+          Assertions.assertThat((String) explain).contains("DeleteFromTableWithFilters");
+          Assertions.assertThat((String) explain).contains("[years(ts) = " + years(date));
+
+          sql(deleteSql);
+
+          List<Object[]> actual = sql(remainSql);
```

Comment:
This should be `SELECT *` without a `WHERE` clause. Otherwise this does not check that the rows were actually removed because they would be ignored by this SQL.